gammapy/datasets/utils.py

```diff
@@ -203,3 +214,165 @@ def split_dataset(dataset, width, margin, split_template_models=True):
                 d.models = dataset.models
             datasets.append(d)
     return datasets
+
+
+def create_map_dataset_from_dl4_irfs(obs, geom=None, energy_axis_true=None, name=None):
```

Comment:
I would recommend to introduce this as another DL4  data format in `MapDataset.read()`. I would check the fermipy documentation again, but ideally this would go as a `yaml` index file, pointing to the file locations and names (however there could be defaults, if fermipy filenames are fixed). Maybe there can be additional support for the single `.npy` session file, that fermipy has.